### ISSUE TITLE
Renaming events from th13 etc to temperaturehumidity1

### ISSUE THREAD

**Issue by basvdijk** [rfxcom/node-rfxcom#67]
Since 1.0.1 events like `th13` etc. don't work anymore and are now handled by `temperaturehumidity1`. I was wondering why did you made this choice? I found working with:

`rfxtrx.on("th13", function (evt) {` 

very convenient instead of 

`rfxtrx.on("temperaturehumidity1", function (evt) {` 

since these are the categories also used by the RFXcom documentation http://www.rfxcom.com/oregon/en

Before 1.0.1 it was easy mapping all the events like `TEMP3` and `TH13` now I have to figure out under which "core event" of this library it appears.

**Comment by maxwellhadley** [comment]
The principle behind the naming is that one RFXCOM packet type corresponds to one event name (if received) and one transmitter object (if transmitted). This is how Kevin originally started out, but it all got a bit diluted when support was added on a piecemeal basis for different meters & sensors, each of which grew their own events. Since the aim, from 1.0.0 on, is to support everything supported by the RFXtrx433, the inconsistent event naming was getting to be a real pain - hence the breaking changes.

I admit the documentation could be improved - maybe use the Wiki?

**Comment by basvdijk** [comment]
Maybe it is an idea to add the protocol to the event data as well? Now you only get the subtype. Adding something like a protocol field like:
```
  protocol:       temperaturehumidity1
  subtype:        13
  id:             0x6103
  seqnbr:         30
  temperature:    5.7
  humidity:       93
  humidityStatus: 3
  batteryLevel:   9
  rssi:           6
```

**Comment by maxwellhadley** [comment]
This came up before - https://github.com/rfxcom/node-rfxcom/pull/31
When I started getting involved with the project some events had a numeric subtype and some had a string, also called 'subtype'. After a bit of discussion we standardised on numeric subtypes everywhere. We could add an additional 'subtypeName' property I suppose if you think it would be helpful. What is your use case for this feature?

**Comment by basvdijk** [comment]
In my home automation system I want to know the device product name as shown at https://github.com/basvdijk/domusto-rfxcom/blob/master/index.ts#L437 This is determined by the the protocol name.

**Comment by maxwellhadley** [comment]
The product name depends on both the packet type and the subtype. The list of supported devices for each subtype changes over time as Bert adds support for new products with each release of the firmware. The subtype identifiers I use are unchanging, only adding new identifiers when a new subtype is added. I suspect you don't want the subtype identifier.

It should be possible to add the supported devices lists somewhere so you could look up by packet type & subtype and get a list of strings, which you could format as you wish (which also makes internationalisation easier, though I admit there is a lot of hardwired English in this package).

Thoughts?

**Comment by maxwellhadley** [comment]
How does this approach sound? For maximum backwards compatibility, I propose to add a second parameter to all received data packet events, which you can access in your handler like this:

    rfxtrx.on("temperaturehumidity1", function (evt, packetType) {..}

where parameter `packetType` is the packet type number, and parameter `evt` is unchanged. You can then retrieve the appropriate device name in your event handler like this:

    names = this.deviceNames[type][evt.subtype];

This will return an array of strings (which will not be empty, although one or more strings in the array may be empty). Each array entry is one of the device names associated with the packet type + subtype combination in the RFXCOM SDK documentation. For example, for a `temperaturehumidity1` event will have `type = 0x54`, and if `evt.subtype = 0x02`, the value of `names` will be `["BTHR918N", "BTHR968"]`. Choose whichever one of these is appropriate.

I have pushed this change to branch [add-device-names](https://github.com/rfxcom/node-rfxcom/tree/add-device-names) if you want to try it out. You will need only files `lib/index.js` and `lib/rfxcom.js`.

**Comment by basvdijk** [comment]
look promising, you now can access the information from the rfxcom plugin instead of self need to manage.

**Comment by maxwellhadley** [comment]
I have merged this change and published to npm as 1.2.0
